### PR TITLE
Add the Gone Hacking Studio XLS

### PR DESCRIPTION
**Pull request by ramonimbao**
## Description

Adds the Gone Hacking Studio XLS numpad.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ghs/xls/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        _______, _______, _______, _______,
+        _______, _______, _______, _______,
+        _______, _______, _______, _______,
+        _______, _______, _______, _______,
+        _______, _______, _______, _______,
+        _______, _______, _______, _______
+    ),
+};
+
+keyevent_t encoder0_cw = {
```
Comment (zvecr) [inline keyboards/ghs/xls/keymaps/via/keymap.c:61]: This is no longer accepted for new contributions, as <https://github.com/qmk/qmk_firmware/pull/17734> provides proper support.

File: keyboards/ghs/xls/info.json [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "encoder": true,
+        "nkro": true
+    },
+    "matrix_pins": {
+        "cols": ["C15", "C14", "C13", "A3"],
+        "rows": ["A4", "B12", "A6", "A10", "A5", "B2", "NO_PIN", "NO_PIN"]
```
Comment (zvecr) [inline keyboards/ghs/xls/info.json:17]:
```suggestion
        "rows": ["A4", "B12", "A6", "A10", "A5", "B2"]
```

File: keyboards/ghs/xls/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,57 @@
+#include QMK_KEYBOARD_H
```
Comment (zvecr) [inline keyboards/ghs/xls/keymaps/default/keymap.c:1]: Licence headers are missing

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by ramonimbao** [comment]
Will reopen a new PR. I messed up rebasing this.

**Pull request closed without merging**